### ISSUE TITLE
This issue is accidentally created, please delete it, I can't do it myself.

### ISSUE THREAD

**Issue by braindotai** [tensorflow/models#7193]
I'm trying to learn Tensorflow 2.0, so I build a toy model and trained it using keres .fit 
method, everything worked well. 
But when I tried to implement the training loop from scratch, the training is happening very very slowly. Keras .fit method trained the model in 1 min 41 secs while the training code I've written taking more than 7 mins to train!!!

Below is my model definition:
```python
model = tf.keras.Sequential()
model.add(layers.Conv2D(filters=6, kernel_size=(3, 3), activation='relu', input_shape=(28, 28, 1)))
model.add(layers.AveragePooling2D())
model.add(layers.Conv2D(filters=16, kernel_size=(3, 3), activation='relu'))
model.add(layers.AveragePooling2D())
model.add(layers.Flatten())
model.add(layers.Dense(units = 120, activation = 'relu'))
model.add(layers.Dense(units = 84, activation = 'relu'))
model.add(layers.Dense(units = 10, activation = 'softmax'))
```
Below I'm defining loss, optimizer and accuracy:
```python
optimizer = tf.keras.optimizers.Adam()
objective = tf.keras.losses.SparseCategoricalCrossentropy()
metric = tf.keras.metrics.SparseCategoricalAccuracy()
```

And Below is my training loop:
```python
%%time
with tf.device('gpu:0'):
    for epoch in range(20):
        cumulative_loss = 0.0
        metric.reset_states()
        for images, labels in dataset:
            with tf.GradientTape() as tape:
                predictions = model(images, training=True)
                loss = objective(labels, predictions)

            grads = tape.gradient(loss, model.trainable_variables)
            optimizer.apply_gradients(zip(grads, model.trainable_variables))

            cumulative_loss += loss
            metric.update_state(labels, predictions)

        print("Epoch: {} Loss: {} Accuracy: {}".format(epoch, cumulative_loss.numpy()/(batch + 1), metric.result()))
```
My Tensorflow GPU Version: 2.0.0-beta1
CUDA Version: 10.0
